index.html

```diff
@@ -25,68 +47,160 @@ <h1 id="result"></h1>
+                    firstOccurence.setDate(firstOccurence.getDate() + 1);
+                }
+                return firstOccurence;
+            }
 
-        // Check if today is the second Wednesday, Thursday, Friday, or Saturday of an even month
-        var isSecondWTFOrSOfEvenMonth = ((day >= 3 && day <= 6) && (date > 7 && date <= 14) && ((month + 1) % 2 === 0));
+            function isGivenDayOfWeekFirstOccurenceAHoliday(dayOfWeek, year, month){
+                // check if the first occurence of the given day of the week in the month is a holiday
+                var firstOccurence = findFirstOccurenceOfGivenDayOfWeek(dayOfWeek, year, month);
+                return isDayAPolishHoliday(firstOccurence);
+            }
 
-        if (isSecondMTOfOddMonth || isSecondWTFOrSOfEvenMonth) {
-            document.getElementById("result").innerHTML = "Tak";
-            document.getElementById("result").classList.add('yes');
-        } else {
-            document.getElementById("result").innerHTML = "Nie";
-            document.getElementById("result").classList.add('no')
-        }
-        var options = { weekday: 'long', year: 'numeric', month: 'long', day: 'numeric' };
-        document.getElementById("date").innerHTML = today.toLocaleDateString("pl-pl", options);
 
-        // Create the calendar
-        var calendar = "<table><tr><th>Pon</th><th>Wto</th><th>Śro</th><th>Czw</th><th>Pią</th><th>Sob</th><th>Nie</th></tr><tr>";
+            function isGierkiDay(fullDate) {
+                var month = fullDate.getMonth();
+                var year = fullDate.getFullYear();
+                var isMonthEven = ((month + 1) % 2 === 0);
+                var dayOfWeek = fullDate.getDay();
+                var dayOfMonth = fullDate.getDate();
 
-        // Fill in the days of the week
-        for (var i = 0; i < dayOfWeek; i++) {
-            calendar += "<td></td>";
-        }
+                if (dayOfWeek === 0){
+                    return false; // nie ma gierek w niedziele
+                }
+                else if (dayOfWeek === 1 || dayOfWeek === 2) {
+                    if (isGivenDayOfWeekFirstOccurenceAHoliday(dayOfWeek, year, month)){
+                        // drugie zajęcia w 3 tygodniu gdy w pierwszym wypada świeto
+                        return !isMonthEven && (dayOfMonth > 14 && dayOfMonth <= 21);
+                    }
+                    else{
+                        return !isMonthEven && (dayOfMonth > 7 && dayOfMonth <= 14);
+                    }
+                }
+                else {
+                    if (isGivenDayOfWeekFirstOccurenceAHoliday(dayOfWeek, year, month)){
+                        // drugie zajęcia w 3 tygodniu gdy w pierwszym wypada świeto
+                        return isMonthEven && (dayOfMonth > 14 && dayOfMonth <= 21);
+                    }
+                    else{
+                        return isMonthEven && (dayOfMonth > 7 && dayOfMonth <= 14);
+                    }
```

Comment:
to brzydkie, bo różni się tylko `!isMonthEven` (gierki pon& wt od gierek w resztę tygodnia). Ja bym dała jakąś zmienną w stylu `evenMonth = isMonthEven if dayOfMWeek in [1, 2] else !isMonthEven` po JS/Pythonowemu i potem tylko jeden blok if - else

Reply:
Brzydkie ale to Twoje ładniejsze by było mniej czytelne jak dla mnie- ta zmienna zamiast "evenMonth" by się musiała nazywać "isMonthEvenParityAdheringToDayOfWeek" - a od samego czytania tej nazwy boli mnie głowa. W tej jakości kodzie te powtórzenia nie bolą aż tak bardzo.

Reply:
No dobra, zrobiłem trochę inaczej ale może rzeczywiście teraz wygląda lepiej.